ts/features/bonus/siciliaVola/screens/voucherGeneration/VoucherGeneratedScreen.tsx

```diff
@@ -11,63 +13,129 @@ import { GlobalState } from "../../../../../store/reducers/types";
-    onPress: props.back,
-    title: "Back"
+    onPress: props.completed,
+    title: I18n.t("global.buttons.exit")
   };
+
+  // TODO: substitute the onPress function with the voucher pdf request
   const continueButtonProps = {
-    primary: false,
-    bordered: true,
-    onPress: props.completed,
-    title: "Esci"
+    primary: true,
+    onPress: () => true,
+    title: I18n.t("global.genericSave")
   };
 
-  // TODO: dispatch props.generateVoucherRequest when the component is mounted
+  if (isNone(maybeVoucherRequest)) {
+    props.failure("Voucher request is None");
+    return null;
+  }
+  const voucherRequest = maybeVoucherRequest.value;
+  if (!isVoucherRequest(voucherRequest)) {
+    props.failure("Partial voucher request is not a complete voucher request");
+    return null;
+  }
 
-  // TODO: manage loading/error/timeout(SvGeneratedVoucherTimeoutScreen) state on generateVoucherRequest selector
-  return (
-    <BaseScreenComponent
-      goBack={true}
-      contextualHelp={emptyContextualHelp}
-      headerTitle={I18n.t("bonus.sv.headerTitle")}
-    >
-      <SafeAreaView
-        style={IOStyles.flex}
-        testID={"VoucherGeneratedScreen"}
-        ref={elementRef}
+  if (!isReady(remoteVoucherGenerated)) {
+    return (
+      <BaseScreenComponent
+        goBack={false}
+        contextualHelp={emptyContextualHelp}
+        headerTitle={I18n.t("bonus.sv.headerTitle")}
       >
-        <ScrollView style={[IOStyles.horizontalContentPadding]}>
-          <H1>{I18n.t("bonus.sv.voucherGeneration.voucherGenerated.title")}</H1>
-        </ScrollView>
-        <FooterWithButtons
-          type={"TwoButtonsInlineHalf"}
-          leftButton={backButtonProps}
-          rightButton={continueButtonProps}
+        <LoadingErrorComponent
+          isLoading={isLoading(remoteVoucherGenerated)}
+          loadingCaption={I18n.t("global.remoteStates.loading")}
+          onRetry={() => props.generateVoucherRequest(voucherRequest)}
         />
-      </SafeAreaView>
-    </BaseScreenComponent>
-  );
+      </BaseScreenComponent>
+    );
+  }
+
+  const voucherGenerated = remoteVoucherGenerated.value;
+
+  switch (voucherGenerated.kind) {
+    // TODO: manage conflict and other state when the final swagger is available
+    case "success":
+      return (
+        <BaseScreenComponent
+          goBack={false}
+          contextualHelp={emptyContextualHelp}
+          headerTitle={I18n.t("bonus.sv.headerTitle")}
+        >
+          <SafeAreaView style={IOStyles.flex} testID={"VoucherGeneratedScreen"}>
+            <ScrollView style={[IOStyles.horizontalContentPadding]}>
+              <H1>
+                {I18n.t("bonus.sv.voucherGeneration.voucherGenerated.title")}
+              </H1>
+              <View spacer />
+              <H5 weight={"Regular"}>
+                {I18n.t("bonus.sv.voucherGeneration.voucherGenerated.subtitle")}
+              </H5>
+              <View spacer />
+              <VoucherInformationComponent
```

Comment:
Not strictly related to this pr, but in case of very long title, there is an overflow of the voucher code + copy button component:
<img width="399" alt="Schermata 2021-10-05 alle 17 02 31" src="https://user-images.githubusercontent.com/26501317/136049342-d74289b1-3f5d-4307-98f9-aad1bf706c39.png">

My suggestion in `VoucherInformationComponent` line `46`:
```typescript
      <H3 color={"bluegreyDark"} style={IOStyles.flex}>
        {props.voucherCode}
      </H3>
```

<img width="403" alt="Schermata 2021-10-05 alle 17 05 00" src="https://user-images.githubusercontent.com/26501317/136049832-a5f47944-3bb7-4b15-b834-e4c19aece3d6.png">